text/0012-asynchronous-updates.md

```diff
@@ -0,0 +1,144 @@
+* Title: Asynchronous Updates
+* Start Date:
+* Specification PR: https://github.com/meilisearch/specifications/pull/12
+* MeiliSearch Issue:
+
+# Asynchronous updates
+
+## 1\. Feature Description and Interaction
+
+### Summary
+
+This spec is the spec of a feature already in place. 
+
+It's scope is:
+- The name for update list and update identifiers. 
+- The route to get one update with this identifier.
+- Te route to get all updates. 
+- The update format (JSON).
+
+### Motivation
+
+MeiliSearch is an asynchronous API. It means that the API does not behave as you would typically expect when handling the request's responses. Some operations are put in a queue and will be executed in turn (asynchronously).
+
+We must define how to interact with this update queue.
+
+### Additional Materials
+
+In Elasticsearch, the updates are not put in an update queue. They put the HTTP client on hold for the indexing process time. This is the case for all types of updates ([Update](https://www.elastic.co/guide/en/elasticsearch/reference/7.x/docs-update.html), [Bulk](https://www.elastic.co/guide/en/elasticsearch/reference/7.x/docs-bulk.html), [Update By Query](https://www.elastic.co/guide/en/elasticsearch/reference/7.x/docs-update-by-query.html), or [Reindex](https://www.elastic.co/guide/en/elasticsearch/reference/7.x/docs-reindex.html)). Only for most of these methods, it is possible to pass an argument (`wait_for_completion=false`) to make updates in [tasks](https://www.elastic.co/guide/en/elasticsearch/reference/7.x/tasks.html). Their task system is quite similar to our update system. Only it is much more agnostic, and it is possible to query the past tasks accurately. 
+
+On the Algolia side, they also call the update queue [tasks](https://www.algolia.com/doc/rest-api/search/#get-a-tasks-status). So they return a `taskID` rather than `updateID` for each request for document or settings updates. 
+
+Other databases don't seem to have an async update queue. 
+
+**TL;DR:**
+- The other solutions call `updates`, `tasks`.
+- Most other solutions allow sending synchronous updates. 
+
+### Explanation
+
+#### Async flow
+
+* When making a written request (create/update/delete) against the search engine, it stores the operation received in a queue and returns a updateId. With this id, the operation update is trackable.
+* Each update received is treated following the order it has been received.
+* You can get the update status on the /updates route.
+* Processed updates are marked as processed and kept in the operation list (available at /indexes/:index\_uid/updates). They won't be deleted.
+
+#### Which operations are async?
+
+Every operation which could be compute-expensive is asynchronous. These include:
+
+* Update index settings
+* Add/update/delete documents
```

Comment:
It would be interesting to also list those that are not

Reply:
Like create an index?

Reply:
Is that all? But yes, overall I think it would be good to   have a good mental model of where the update queue stands when interacting with an index documented somewhere, wdyt?

Reply:
Yes, updated!